include/wpe/file-system.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2022 Igalia S.L.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if !defined(__WPE_H_INSIDE__) && !defined(WPE_COMPILATION)
+#error "Only <wpe/wpe.h> can be included directly."
+#endif
+
+#ifndef wpe_file_system_h
+#define wpe_file_system_h
+
+/**
+ * SECTION:file-system
+ * @short_description: Library File System
+ * @title: File System
+ */
+
+#if defined(WPE_ENABLE_FS) && WPE_ENABLE_FS
+
+#if defined(WPE_COMPILATION)
+#include "export.h"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum wpe_file_system_path {
+    WPE_FILE_SYSTEM_PATH_USER_DATA,
+    WPE_FILE_SYSTEM_PATH_USER_CACHE,
+};
```

Comment:
Need to see if there are more relevant paths to add here.

Reply:
I think it's fine to start with these two, then add other needed paths as needed later on.

For example I imagine this might come in handy to inform where to looks for the injected bundle when running on Android, currently we have to [patch WebKit and sacrifice a goat](https://github.com/Igalia/wpe-android/pull/57) to get it loaded.